test/e2e/vsphere_volume_placement.go

```diff
@@ -0,0 +1,225 @@
+)
+
+var _ = framework.KubeDescribe("vsphere-volume-placement", func() {
+	f := framework.NewDefaultFramework("vsphere-volume-placement")
+	var (
+		c                  clientset.Interface
+		ns                 string
+		vsp                *vsphere.VSphere
+		volumePath         string
+		node1Name          string
+		node1LabelValue    string
+		node1KeyValueLabel map[string]string
+
+		node2Name          string
+		node2LabelValue    string
+		node2KeyValueLabel map[string]string
+
+		isNodeLabeled bool
+	)
+	BeforeEach(func() {
+		framework.SkipUnlessProviderIs("vsphere")
+		c = f.ClientSet
+		ns = f.Namespace.Name
+		framework.ExpectNoError(framework.WaitForAllNodesSchedulable(c, framework.TestContext.NodeSchedulableTimeout))
+
+		By("creating vmdk")
+		vsp, err := vsphere.GetVSphere()
+		Expect(err).NotTo(HaveOccurred())
+
+		volumePath, err = createVSphereVolume(vsp, nil)
+		Expect(err).NotTo(HaveOccurred())
+
+		if !isNodeLabeled {
+			nodeList := framework.GetReadySchedulableNodesOrDie(f.ClientSet)
+			if len(nodeList.Items) != 0 {
+				node1Name = nodeList.Items[0].Name
+				node2Name = nodeList.Items[1].Name
+			} else {
+				framework.Failf("Unable to find ready and schedulable Node")
+			}
+			node1LabelValue = "vsphere_e2e_" + string(uuid.NewUUID())
+			node1KeyValueLabel = make(map[string]string)
+			node1KeyValueLabel["vsphere_e2e_label"] = node1LabelValue
+			framework.AddOrUpdateLabelOnNode(c, node1Name, "vsphere_e2e_label", node1LabelValue)
+
+			node2LabelValue = "vsphere_e2e_" + string(uuid.NewUUID())
+			node2KeyValueLabel = make(map[string]string)
+			node2KeyValueLabel["vsphere_e2e_label"] = node2LabelValue
+			framework.AddOrUpdateLabelOnNode(c, node2Name, "vsphere_e2e_label", node2LabelValue)
+
+		}
+
+	})
+
+	AddCleanupAction(func() {
+		if len(node1LabelValue) > 0 {
+			framework.RemoveLabelOffNode(c, node1Name, "vsphere_e2e_label")
+		}
+		if len(node2LabelValue) > 0 {
+			framework.RemoveLabelOffNode(c, node2Name, "vsphere_e2e_label")
+		}
+		if len(volumePath) > 0 {
+			vsp, err := vsphere.GetVSphere()
+			Expect(err).NotTo(HaveOccurred())
+			vsp.DeleteVolume(volumePath)
+		}
+	})
+
+	/*
+		Test Steps
+		-----------
+		1. Find node with the status available and ready for scheduling.
+		2. Add label to the node. - (vsphere_e2e_label: Random UUID)
+		3. Create VMDK volume
+		4. Create POD Spec with volume path of the vmdk and NodeSelector set to label assigned in the Step 2.
+		5. Create POD and wait for POD to become ready.
+		6. Verify volume is attached to the node labeled in the Step 2.
+		7. Delete POD.
+		8. Wait for volume to be detached from the node.
+		9. Repeat Step 5 to 7 and make sure back to back pod creation with the same volume is working as expected.
+		10. Delete VMDK volume
+	*/
+	framework.KubeDescribe("Test Back-to-back pod creation/deletion with the same volume source on the same worker node", func() {
+		It("should provision pod on the node with matching label", func() {
```

Comment:
Yes this looks good. We have other similar tests, I am planning to add in next PR. will just keep adding more `It`s to this `KubeDescribe`